app/uk/gov/hmrc/tai/model/domain/TaxOnOtherIncome.scala

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.tai.model.domain
+
+import play.api.libs.json.{JsResult, JsSuccess, JsValue, Reads}
+import uk.gov.hmrc.tai.model.domain.NpsIabdSummary.totalLiabilityIabds
+import uk.gov.hmrc.tai.model.domain.RateBand.incomeAndRateBands
+
+import scala.annotation.tailrec
+
+case class TaxOnOtherIncome(tax: BigDecimal)
+
+object TaxOnOtherIncome {
+  private val NonCodedIncome = 19
+
+  val taxOnOtherIncomeRead = new Reads[Option[BigDecimal]] {
```

Comment:
There are two reads with almost identical names in here. This one just reads the value and the other one at line 35 seems to do a lot more. I wonder if we could find better names for these?

Reply:
I would not touch the naming just yet

Reply:
OK we can leave for now